UnitySDK/Assets/ML-Agents/Scripts/LearningBrain.cs

```diff
@@ -58,152 +42,45 @@ public class LearningBrain : Brain
         /// <param name="communicator"> The Batcher the brain will use for the current session</param>
         private void SetCommunicator(ICommunicator communicator)
         {
-            m_Communicator = communicator;
-            m_Communicator?.SubscribeBrain(name, brainParameters);
+            m_BatchedDecisionMaker = communicator;
+            communicator?.SubscribeBrain(name, brainParameters);
             LazyInitialize();
 
         }
 
         /// <inheritdoc />
         protected override void Initialize()
         {
-            ReloadModel();
-            var comm = FindObjectOfType<Academy>()?.Communicator;
+            var aca = FindObjectOfType<Academy>();
+            var comm = aca?.Communicator;
             SetCommunicator(comm);
-        }
-
-        /// <summary>
-        /// Initializes the Brain with the Model that it will use when selecting actions for
-        /// the agents
-        /// </summary>
-        /// <param name="seed"> The seed that will be used to initialize the RandomNormal
-        /// and Multinomial obsjects used when running inference.</param>
-        /// <exception cref="UnityAgentsException">Throws an error when the model is null
-        /// </exception>
-        public void ReloadModel(int seed = 0)
-        {
-            if (m_TensorAllocator == null)
-                m_TensorAllocator = new TensorCachingAllocator();
-
-            if (model != null)
+            if (aca == null || comm != null)
             {
-#if BARRACUDA_VERBOSE
-                _verbose = true;
-#endif
-
-                D.logEnabled = m_Verbose;
-
-                // Cleanup previous instance
-                if (m_Engine != null)
-                    m_Engine.Dispose();
-
-                m_BarracudaModel = ModelLoader.Load(model.Value);
-                var executionDevice = inferenceDevice == InferenceDevice.GPU
-                    ? BarracudaWorkerFactory.Type.ComputePrecompiled
-                    : BarracudaWorkerFactory.Type.CSharp;
-
-                m_Engine = BarracudaWorkerFactory.CreateWorker(executionDevice, m_BarracudaModel, m_Verbose);
+                return;
             }
-            else
+            var modelRunner = aca.ModelRunners.Find(x => x.HasModel(model));
```

Comment:
This feels a little weird that LearningBrain is the only thing that accesses the public member on Academy. Maybe move this logic to something like `Academy.GetOrCreateModelRunner()`? Not sure I love that either though.

Reply:
Any other suggestions ?

Reply:
How do you feel about moving it to Academy? I guess the counter argument is that it requires passing the brainParameters and inferenceDevice too, like

``` csharp
ModelRunner GetOrCreateModelRunner(NNModel model, BrainParameters brainParameters, InferenceDevice inferenceDevice) {...}
```
@surfnerd any preference?

Reply:
I updated the code. Let me know what you think.

Reply:
I think for now what you have looks fine. We can revisit in a follow up of the Academy cleanup.